utils/bfd.ml

```diff
@@ -0,0 +1,540 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                        Nicolas Ojeda Bar, LexiFi                       *)
+(*                                                                        *)
+(*   Copyright 2020 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+type error =
+  | Truncated_file
+  | Unrecognized of string
+  | Unsupported of string * int
+
+exception Error of error
+
+let name_at ?max buf start =
+  assert (start >= 0 && start <= Bytes.length buf);
```

Comment:
Wouldn't this assert be better as an explicit bounds check followed by a `raise (Error ...)` on failure? It's more than an internal invariant, and the error will be an inscrutable bounds check error from within the function if assertions are disabled.

Reply:
Yes, it would be. Will amend.